test/types/hooks.test-d.ts

```diff
@@ -213,11 +219,15 @@ server.addHook('onClose', async (instance) => {
 // Use case to monitor any regression on issue #3620
 // ref.: https://github.com/fastify/fastify/issues/3620
 const customTypedHook: preHandlerAsyncHookHandler<
-RawServerBase,
+http.Server,
 RawRequestDefaultExpression,
 RawReplyDefaultExpression,
-Record<string, unknown>
-> = async function (request, reply) {
+RouteGenericInterface,
+ContextConfigDefault,
+FastifySchema,
+FastifyTypeProviderDefault,
+ResolveFastifyRequestType<FastifyTypeProviderDefault, FastifySchema, RouteGenericInterface>
+> = async function (request, reply): Promise<void> {
```

Comment:
Why this change?

Reply:
Seems like when using `RawServerBase` type I get the following error: 
```ts
  test/types/hooks.test-d.ts:231:30
  ✖  231:30  Argument of type FastifyInstance<RawServerBase, IncomingMessage, ServerResponse, FastifyLoggerInstance, FastifyTypeProviderDefault> is not assignable to parameter of type FastifyInstance<Server, IncomingMessage, ServerResponse, FastifyLoggerInstance, FastifyTypeProviderDefault>.
  Types of property server are incompatible.
    Type RawServerBase is not assignable to type Server.  
```

Puzzles me a bit since I did not change anything related of the server type. Since we got an alias type for `http.Server` I switched to use now: `RawServerDefault`

Reply:
cc @fastify/typescript

Reply:
I've cloned this and looked into it. Actually, I think that @MarcoLeko discovered some strange bug with the `FastifyInstance` generic type. This stopped working because, in `preHandlerAsyncHookHandler`, we defined this as `FastifyInstance` without passing any generic. So @MarcoLeko passed correct generics derived from `preHandlerAsyncHookHandler,` and now the error is thrown.
current: https://github.com/fastify/fastify/blob/main/types/hooks.d.ts#L165
proposed: https://github.com/MarcoLeko/fastify/blob/bugfix/streamline-fastify-request-to-be-generic/types/hooks.d.ts#L185

@fastify/typescript any ideas?

Reply:
Ok, never mind. I understand what is going on. `RawServerBase` is a union type, and we should not use it to test our types! We should pick one of the types of union! `RawServerDefault` should be used as the test case!